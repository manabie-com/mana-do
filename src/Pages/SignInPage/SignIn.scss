@import url('https://fonts.googleapis.com/css2?family=Noto+Sans:wght@700&family=Poppins:wght@400;500;600&display=swap');

.SignIn {
    text-align: center;
    font-family: "Poppins", sans-serif;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 400px;
    background: white;
    border-radius: 8px;
    box-shadow: 10px 10px 15px rgba(0, 0, 0, 0.05);
}

.SingIn h1 {
    text-align: center;
    padding: 0 0 20px 0;
    border-bottom: 1px solid silver;

}

.SingIn .form {
    padding: 0 40px;
    box-sizing: border-box;
}

form .txt_field {
    position: relative;
    margin: 15px 0;
    display: inline-flex;

}

.txt_field input {
    width: 300px;
    height: 40px;
    padding: 0 5px;
    font-size: 16px;
    border: none;
    background: none;
    outline: none;
}
