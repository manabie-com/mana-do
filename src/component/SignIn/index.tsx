import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import Service from "service";

const LoginForm = () => {
  const [form, setForm] = useState({
    userId: "",
    password: "",
  });
  const history = useHistory();
  const [error, setError] = useState({
    msg: "",
  });

  const onSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    try {
      const resp = await Service.signIn(form.userId, form.password);
      localStorage.setItem("token", resp);
      history.push("/todo");
      setError({ msg: "" });
    } catch (error) {
      setError({
        msg: "Incorrect Username/Password",
      });
    }
  };

  const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.persist();
    setForm((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
    setError({ msg: "" });
  };

  return (
    <div className="login">
      <h1 className="login__heading">Log In to ToDo</h1>

      <form
        onSubmit={onSubmit}
        className="login__form"
        data-testid="login-form"
      >
        <label htmlFor="user_id" className="login__form-control">
          <span>User id</span>
          <input
            id="user_id"
            name="userId"
            value={form.userId}
            onChange={onChangeField}
            data-testid="userId"
          />
        </label>
        <br />
        <label htmlFor="password" className="login__form-control">
          <span>Password</span>
          <input
            id="password"
            name="password"
            type="password"
            value={form.password}
            onChange={onChangeField}
          />
        </label>
        <br />
        <button className="login__button" type="submit" data-testid="submit">
          Sign in
        </button>
      </form>
      {error.msg && (
        <p data-testid="error-msg" className="login__error-msg">
          {error.msg}
        </p>
      )}
    </div>
  );
};

export default LoginForm;
