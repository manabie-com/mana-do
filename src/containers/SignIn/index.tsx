import React, { useState } from 'react';

import { useHistory } from 'react-router-dom';
import Service from '../../service';
import './style.scss';
import background from '../../assets/images/background.png';
import Input from '../../components/Input';
import Button from '../../components/Button';
const prefix = 'sign-in';
const SignInContainer = () => {
  //state
  const [form, setForm] = useState({
    userId: '',
    password: '',
  });
  const [state, setState] = useState({
    msgErr: null
  });
  
  const history = useHistory();
  //function 
  async function signIn(e:React.FormEvent<HTMLFormElement>) {
    e.preventDefault();
    await Service.signIn(form.userId, form.password)
      .then((resp) => {
        if (resp) {
          localStorage.setItem('token', resp);
          history.push('/todo');
        }
      })
      .catch((err) => 
      {
        setState({...state, msgErr: err})
      });
  };
  const onChangeField = (e:React.ChangeEvent<HTMLInputElement>) => {
    e.persist();
    setForm((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };
  return (
    <div className={prefix}>
      <div className={`${prefix}__logo`}>
        <img src={background} alt="none" />
      </div>
      <div className={`${prefix}__content`}>
        <form onSubmit={signIn} className={`${prefix}__content__form`}>
          <div className={`${prefix}__content__form__item`}>
            <Input
              label="User ID"
              id="user_id"
              name="userId"
              value={form.userId}
              onChange={onChangeField}
            />
          </div>
          <div className={`${prefix}__content__form__item`}>
            <Input
              label="Password"
              id="password"
              name="password"
              type="password"
              value={form.password}
              onChange={onChangeField}
            />
          </div>
          {state.msgErr && <span className="isErr">{state.msgErr}</span>}
          <div className={`${prefix}__content__form__item`}>
            <Button type="submit" size="large" colorType="success"> Sign in</Button>
          </div>
        </form>
      </div>
    </div>
  );
};

export default SignInContainer;
