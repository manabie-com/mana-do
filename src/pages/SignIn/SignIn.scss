.SigIn {
    box-shadow: 0 0 70px -20px rgba(0, 0, 0, 0.5);
    background-color: #fff;
    border-radius: 10px;
    padding: 1rem 2rem;
    display: flex;
    align-items: center;
    justify-content: center;
    min-width: 400px;
    height: max-content;
    margin: auto;
    &__Container {
        width: 100%;
    }
    h1 {
        text-align: center;
        text-transform: uppercase;
    }
    form {
        .form-group {
            display: flex;
            align-items: center;
            margin-bottom: 1rem;
            label {
                margin-right: 1rem;
                flex-basis: 25%;
                text-align: left;
            }
            input {
                flex-basis: 75%;
                width: 100%;
                border: 1px solid #ccc;
            }
        }
        button {
            margin: 1rem 0 2rem;
            font-size: 1.3rem;
            padding: 0.5rem 2rem;
            background-color: #0d6efd;
            color: #fff;
            cursor: pointer;
            transition: 0.4s all ease;
        }
        p {
            margin-bottom: 1rem;
            color: red;
        }
    }
}
