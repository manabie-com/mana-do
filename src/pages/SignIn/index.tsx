import React, { useState } from "react";

import { useHistory } from "react-router-dom";
import Service from "../../service";

import { Button, Input } from "../../components";

const SignInPage = () => {
  const [form, setForm] = useState({
    userId: "",
    password: "",
  });
  const [errorMessage, setErrorMessage] = useState("");
  const history = useHistory();

  const signIn = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    try {
      const resp = await Service.signIn(form.userId, form.password);

      localStorage.setItem("token", resp);
      history.push("/");
    } catch (error) {
      setErrorMessage(error);
    }
  };

  const onChangeField = (e: React.ChangeEvent<HTMLInputElement>) => {
    e.persist();
    setForm((prev) => ({
      ...prev,
      [e.target.name]: e.target.value,
    }));
  };

  return (
    <div className="content-container">
      <h1 data-testid="header">Sign In</h1>
      {errorMessage && <div className="fail">{errorMessage}</div>}
      <form onSubmit={signIn}>
        <Input
          variant="text"
          name="userId"
          label="UserId"
          placeholder="Enter your user id"
          valueExternal={form.userId}
          style={{ marginTop: 12 }}
          onChange={onChangeField}
        />
        <Input
          variant="password"
          name="password"
          label="Password"
          placeholder="Enter your password"
          valueExternal={form.password}
          style={{ marginTop: 12 }}
          onChange={onChangeField}
        />
        <hr />
        <Button
          type="submit"
          testId="btnSignin"
          classNames="btn__primary"
          disabled={!form.userId || !form.password}
          text="Sign in"
        />
      </form>
    </div>
  );
};

export default SignInPage;
