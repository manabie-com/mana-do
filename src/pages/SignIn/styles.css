section {
  position: relative;
  background-color: #112d42;
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100vh;
}
section::before {
  content: '';
  position: absolute;
  left: 0;
  top: 0;
  background-color: #03a9f4;
  width: 50%;
  height: 100%;
}
.form-contact {
  position: relative;
  width: 1100px;
  height: 550px;
  display: flex;
  z-index: 10;
  align-items: center;
}
.contactinfo {
  position: absolute;
  width: 350px;
  /* height: calc(100%-80px); */
  /* height: 70vh; */
  background-color: #0f3959;
  z-index: 6;
  padding: 0px 40px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  box-shadow: 0 20px 20px rgba(0, 0, 0, 0.2);
}
.contactinfo h2 {
  color: white;
  font-size: 24px;
  font-weight: 500;
}
.contactinfo .info {
  position: relative;
  margin: 20px 0;
}
.contactinfo .info li {
  position: relative;
  list-style: none;
  display: flex;
  margin: 20px 0;
  cursor: pointer;
  align-items: flex-start;
}
.contactinfo .info li span:nth-child(1) {
  width: 30px;
  min-width: 30px;
}
.contactinfo .info li span:nth-child(1) i {
  min-width: 100%;
  filter: invert(1);
  opacity: 0.5;
}
.contactinfo .info li span:nth-child(2) {
  color: white;
  margin-left: 10px;
  font-weight: 300;
  opacity: 0.5;
}
.contactinfo .info li:hover span:nth-child(1) i,
.contactinfo .info li:hover span:nth-child(2) {
  opacity: 1;
}
.contactinfo .sci {
  position: relative;
  padding: 0;
  display: flex;
  justify-content: center;
}
.contactinfo .sci li {
  list-style: none;
  margin-left: 15px;
}
.contactinfo .sci li a {
  text-decoration: none;
}
.contactinfo .sci li a i {
  color: black;
  filter: invert(1);
  opacity: 0.5;
}
.contactinfo .sci li:hover a i {
  opacity: 1;
}

/* phan 2 */
.contactform {
  position: absolute;
  padding: 70px 50px;
  background-color: white;
  margin-left: 250px;
  padding-left: 250px;
  width: calc(100%-150px);
  height: 65vh;
  box-shadow: 0 50px 50px rgba(0, 0, 0, 0.5);
}
.contactform a {
  text-decoration: none;
  color: black;
}
.contactform a:hover {
  color: tomato;
}
.contactform h2 {
  color: #0f3959;
  font-size: 24px;
  font-weight: 500;
}
.contactform .formBox {
  position: relative;
  display: flex;
  justify-content: space-between;
  flex-wrap: wrap;
  padding-top: 30px;
}
.contactform .formBox .inputBox {
  position: relative;
  margin: 0 0 35px 0;
}
.contactform .formBox .inputBox.w50 {
  width: 47%;
}
.contactform .formBox .inputBox.w100 {
  width: 100%;
}
.contactform .formBox .inputBox input,
.contactform .formBox .inputBox textarea {
  width: 100%;
  padding: 5px 0;
  font-size: 18px;
  font-weight: 300;
  color: #333;
  border: none;
  border-bottom: 1px solid #777;
  outline: none;
  resize: none;
}
.contactform .formBox .inputBox textarea {
  min-height: 120px;
}
.contactform .formBox .inputBox span {
  position: absolute;
  left: 0;
  padding: 5px 0;
  font-size: 18px;
  font-weight: 300;
  color: #333;
  transition: 0.5s;
  pointer-events: none;
}
.contactform .formBox .inputBox p {
  color: red;
  font-size: 12px;
  margin: 2px;
}
.contactform .formBox .inputBox input:focus ~ span,
.contactform .formBox .inputBox textarea:focus ~ span,
.contactform .formBox .inputBox input:valid ~ span,
.contactform .formBox .inputBox textarea:valid ~ span {
  transform: translateY(-20px);
  font-size: 12px;
  font-weight: 400;
  letter-spacing: 1px;
  color: tomato;
}
.contactform .formBox .inputBox input[type='submit'] {
  position: relative;
  cursor: pointer;
  background-color: #0f3959;
  color: white;
  border: none;
  max-width: 150px;
  padding: 12px;
}
.contactform .formBox .inputBox input[type='submit']:hover {
  background-color: tomato;
}
