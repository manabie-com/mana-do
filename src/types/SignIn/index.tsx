import { Dispatch } from "redux"

export interface SignInProps {
    dispatch: Dispatch
}