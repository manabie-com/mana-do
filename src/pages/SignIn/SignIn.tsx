import React, { useState } from 'react';

import { useHistory } from 'react-router-dom'
import Service from '../../service'

import "./SignIn.scss";

const SignIn = () => {
    const [form, setForm] = useState({
        userId: '',
        password: ''
    });
    const [err, setErr] = useState<string>("")
    const history = useHistory();

    const signInHandler = async (e: React.FormEvent<HTMLFormElement>) => {
        e.preventDefault()
        try {
            const resp = await Service.signIn(form.userId, form.password)
            localStorage.setItem('token', resp);
            history.push('/todo')
        } catch (error) {
            setErr(error)
        }
    }

    const onChangeFieldHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
        e.persist()
        setForm(prev => ({
            ...prev,
            [e.target.name]: e.target.value
        }))
    }

    return (
        <div className="SigIn">
            <div className="SigIn__Container">
                <h1>Log in</h1>
                <form onSubmit={signInHandler}>
                    <div className="form-group">
                        <label htmlFor="user_id">User id</label>
                        <input
                            id="user_id"
                            name="userId"
                            type="text"
                            value={form.userId}
                            autoComplete="off"
                            onChange={onChangeFieldHandler}
                        />
                    </div>

                    <div className="form-group">
                        <label htmlFor="password" >Password</label>
                        <input
                            id="password"
                            name="password"
                            type="password"
                            value={form.password}
                            onChange={onChangeFieldHandler}
                        />
                    </div>

                    <button type="submit">
                        Sign in
                    </button>
                    {err && <p>{err}</p>}
                </form>
                <p>Log in with: firstUser/example</p>
            </div>
        </div>
    );
};

export default SignIn;