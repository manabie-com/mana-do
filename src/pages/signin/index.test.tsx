import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { shallow } from 'enzyme';
import SignIn from './index';

describe('<Signin /> ', () => {
  test('Render title', () => {
    const { getByText } = render(<SignIn />);
    const title = getByText(/Welcome/i);

    expect(title).toBeInTheDocument();
  });

  test('Render input user id', () => {
    const { getByPlaceholderText } = render(<SignIn />);
    const inputUserId = getByPlaceholderText('Please input your user id');
    expect(inputUserId).toBeInTheDocument();
  });

  test('Render input password', () => {
    const { getByPlaceholderText } = render(<SignIn />);
    const inputPassword = getByPlaceholderText('Please input your password');
    expect(inputPassword).toBeInTheDocument();
  });

  test('Render button signin', () => {
    const { getByText } = render(<SignIn />);
    const buttonSignin = getByText('Sign in');
    expect(buttonSignin).toBeInTheDocument();
  });

  const setupInput = () => {
    const utils = render(<SignIn />);
    const input = utils.getByLabelText('user_id');
    return { input, ...utils };
  };

  test('It should change value onChange input', () => {
    const { input } = setupInput();
    expect(input.value).toBe('');
    fireEvent.change(input, { target: { value: 'firstuser' } });
    expect(input.value).toBe('firstuser');
  });

  test('Match snapshot', () => {
    const wrapper = shallow(<SignIn />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should handle click submit button', () => {
    const fn = jest.fn();
    const utils = render(<SignIn />);
    const submit = utils.getByText('Sign in');
    fireEvent.click(submit);
  });
});
