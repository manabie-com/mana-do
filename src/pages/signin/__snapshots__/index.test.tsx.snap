// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin />  ..... 1`] = `ShallowWrapper {}`;

exports[`<Signin />  Match snapshot 1`] = `ShallowWrapper {}`;
